semanticdb/semanticdb3/javacp.md

```diff
@@ -0,0 +1,480 @@
+### Java
+
+In this section, we exhaustively map Java language features onto SemanticDB.
+As a reference, we use the [JVM Specification][jvms] (referred to as "JVMS" in the text below)
+and [Java Language Specification][jls] (referred to as "JLS" in the text below).
+
+<a name="java-symbol"></a>
+#### Symbol
+
+In this section, we describe the Java symbol format.
+
+<table>
+  <tr>
+    <td><b>Symbols</b></td>
+    <td><b>Format</b></td>
+  </tr>
+  <tr>
+    <td>
+      Global symbols
+      <a href="#symbol">↑</a>
+    </td>
+    <td>
+      <ul>
+        <li>
+          For root package <a href="https://www.scala-lang.org/files/archive/spec/2.12/09-top-level-definitions.html#package-references">[20]</a>,
+          its descriptor.
+        </li>
+        <li>
+          For empty package <a href="https://www.scala-lang.org/files/archive/spec/2.12/09-top-level-definitions.html#packagings">[21]</a>,
+          its descriptor.
+        </li>
+        <li>
+          For top-level package, its descriptor.
+        </li>
+        <li>
+          For other definition, concatenation of owner symbol and
+          definition descriptor.
+        </li>
+      </ul>
+    </td>
+  </tr>
+</table>
+
+Note that Java symbols do not have [local symbols](#symbol), unlike the [Scala symbol format](#scala-symbol).
+
+**Owner** is:
+  * For root package, `None`.
+  * For empty package, root package.
+  * For top-level package, root package.
+  * For other package, parent package.
+  * For other top-level definition, its package.
+  * For other other global definition, the innermost enclosing definition.
+  * For other definitions, `None`.
+
+**Descriptor** is:
+  * For `LOCAL`, `MACRO`, `OBJECT`, `TRAIT`, `SELF_PARAMETER`, `TYPE` or `PACKAGE_OBJECT`, unsupported.
```

Comment:
`MACRO`? `OBJECT`? I don't think that here we should talk about kinds that aren't realizable in Java.